pkg/client-nfd/compat/node-validator/node-validator.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodevalidator
+
+import (
+	"context"
+	"slices"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/registry"
+
+	nfdv1alpha1 "sigs.k8s.io/node-feature-discovery/api/nfd/v1alpha1"
+	"sigs.k8s.io/node-feature-discovery/pkg/apis/nfd/nodefeaturerule"
+	compatspec "sigs.k8s.io/node-feature-discovery/pkg/client-nfd/compat/spec"
+	"sigs.k8s.io/node-feature-discovery/source"
+
+	// register sources
+	_ "sigs.k8s.io/node-feature-discovery/source/cpu"
+	_ "sigs.k8s.io/node-feature-discovery/source/kernel"
+	_ "sigs.k8s.io/node-feature-discovery/source/memory"
+	_ "sigs.k8s.io/node-feature-discovery/source/network"
+	_ "sigs.k8s.io/node-feature-discovery/source/pci"
+	_ "sigs.k8s.io/node-feature-discovery/source/storage"
+	_ "sigs.k8s.io/node-feature-discovery/source/system"
+	_ "sigs.k8s.io/node-feature-discovery/source/usb"
+)
+
+type Args struct {
+	RegReference *registry.Reference
+	Tags         []string
+	Platform     *ocispec.Platform
+	PlainHTTP    bool
+}
+
+type nodeValidator struct {
+	Args Args
+}
+
+func New(opts ...NodeValidatorOpts) nodeValidator {
```

Comment:
it's same effect to initialize a new NodeValidator without introduce extra nodeNolidatorOpts

```
func New2(args *Args) NodeValidator {
	return NodeValidator{
		Args: *args,
	}
}
```

usually a pointer is preferred to return in initialization, like 

```
func New3(args *Args) *NodeValidator {
	return &NodeValidator{
		Args: *args,
	}
}
```

Reply:
It could work like this, although, the other commands (like `nfd-master` and `nfd-worker`) implement the options pattern. As I was asked, I am trying to stay as close as possible to other commands initialization.